pystapi-client/src/pystapi_client/client.py

```diff
@@ -264,26 +273,97 @@ def get_product(self, product_id: str) -> Product:
 
         Returns:
             Product: A STAPI Product
+        """
+        product_endpoint = self._get_products_href(product_id)
+        product_json = self.stapi_io.read_json(endpoint=product_endpoint)
+        return Product.model_validate(product_json)
+
+    def get_product_opportunities(
+        self,
+        product_id: str,
+        date_range: tuple[str, str],
+        geometry: dict[str, Any],
+        cql2_filter: CQL2Filter | None = None,
+        limit: int = 10,
+    ) -> Iterator[OpportunityCollection]:  # type: ignore[type-arg]
```

Comment:
Same as above, I think we should break this down to return a `Iterator[Opportunity]`.